### PR TITLE
[bsp][lpc55sxx]add: rw007_lpc55sxx_port.c

### PR DESCRIPTION
**Pull request by Z8MAN8**
## 拉取/合并请求描述：(PR description)

[

#### 在什么测试环境下测试通过 (what is the test environment)

]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [ ] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [ ] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [ ] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [ ] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [ ] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [ ] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 代码是高质量的 Code in this PR is of high quality
- [ ] 已经使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

File: bsp/lpc55sxx/lpc55s69_nxp_evk/board/ports/rw007_lpc55sxx_port.c [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#include <rtthread.h>
```
Comment (supperthomas) [inline bsp/lpc55sxx/lpc55s69_nxp_evk/board/ports/rw007_lpc55sxx_port.c:1]: license

**Review by supperthomas — CHANGES_REQUESTED** [review]

**Review by supperthomas — APPROVED** [review]

File: bsp/lpc55sxx/lpc55s69_nxp_evk/board/ports/rw007_lpc55sxx_port.c [hunk 2]
```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2006-2023, RT-Thread Development Team
```
Comment (mysterywolf) [inline bsp/lpc55sxx/lpc55s69_nxp_evk/board/ports/rw007_lpc55sxx_port.c:2]: rw007_port.c

File: bsp/lpc55sxx/lpc55s69_nxp_evk/board/ports/rw007_lpc55sxx_port.c [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    spi_wifi_isr(0);
+}
+
+void spi_wifi_hw_init(void)
+{
+    rt_pin_attach_irq(RW007_INT_BUSY_PIN, PIN_IRQ_MODE_FALLING, int_wifi_irq, 0);
+    rt_pin_irq_enable(RW007_INT_BUSY_PIN, RT_TRUE);
+}
+
+
```
Comment (mysterywolf) [inline bsp/lpc55sxx/lpc55s69_nxp_evk/board/ports/rw007_lpc55sxx_port.c:77]:
```suggestion
```

File: bsp/lpc55sxx/lpc55s69_nxp_evk/board/ports/rw007_port.c [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+{
+    ((void)p);
+    spi_wifi_isr(0);
+}
+
+void spi_wifi_hw_init(void)
+{
+    rt_pin_attach_irq(RW007_INT_BUSY_PIN, PIN_IRQ_MODE_FALLING, int_wifi_irq, 0);
+    rt_pin_irq_enable(RW007_INT_BUSY_PIN, RT_TRUE);
+}
```
Comment (mysterywolf) [inline bsp/lpc55sxx/lpc55s69_nxp_evk/board/ports/rw007_port.c:75]:
```suggestion
}

```

File: bsp/lpc55sxx/lpc55s69_nxp_evk/board/SConscript [hunk 5]
```diff
@@ -9,6 +9,9 @@ MCUX_Config/board/clock_config.c
 MCUX_Config/board/pin_mux.c
 """)
 
+if GetDepend(['PKG_USING_RW007']):
+    src += Glob('ports/rw007_port.c')
```
Comment (mysterywolf) [inline bsp/lpc55sxx/lpc55s69_nxp_evk/board/SConscript:13]:
```suggestion
    src += ['ports/rw007_port.c']
```

File: bsp/lpc55sxx/lpc55s69_nxp_evk/board/SConscript [hunk 6]
```diff
@@ -9,6 +9,9 @@ MCUX_Config/board/clock_config.c
...
+
 CPPPATH = [cwd,cwd + '/MCUX_Config/board']
```
Comment (mysterywolf) [inline bsp/lpc55sxx/lpc55s69_nxp_evk/board/SConscript:15]:
```suggestion
CPPPATH = [cwd, cwd + '/MCUX_Config/board']
```

Comment (mysterywolf) [reply bsp/lpc55sxx/lpc55s69_nxp_evk/board/SConscript:15]: os.path.join(cwd, 'MCUX_Config', 'board')

File: bsp/lpc55sxx/lpc55s69_nxp_evk/board/ports/rw007_port.c [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+    rt_thread_delay(rt_tick_from_millisecond(200));
+    rt_pin_mode(RW007_INT_BUSY_PIN, PIN_MODE_INPUT_PULLUP);
+}
+
+int wifi_spi_device_init(void)
+{
+    char sn_version[32];
+
+    rt_err_t rt_hw_spi_device_attach(const char *bus_name, const char *device_name, rt_uint32_t pin);
```
Comment (mysterywolf) [inline bsp/lpc55sxx/lpc55s69_nxp_evk/board/ports/rw007_port.c:47]: include <drv_spi.h>

**Review by mysterywolf — CHANGES_REQUESTED** [review]

**Review by mysterywolf — APPROVED** [review]

**Pull request merged**